runtime/jcl/common/java_dyn_methodhandle.c

```diff
@@ -1045,3 +1046,47 @@ clearNonZAAPEligibleBit(JNIEnv *env, jclass nativeClass, const JNINativeMethod *
 	vmFuncs->internalExitVMToJNI(vmThread);
 }
 #endif /* J9VM_OPT_JAVA_OFFLOAD_SUPPORT */
+
+#if JAVA_SPEC_VERSION >= 15
+void JNICALL
+Java_java_lang_invoke_MethodHandleNatives_checkClassBytesImpl(JNIEnv *env, jclass jlClass, jbyteArray classRep)
+{
+	J9VMThread *currentThread = (J9VMThread *)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	jsize length = 0;
+	U_8* classBytes = NULL;
+	I_32 rc = 0;
+	U_8* segment = NULL;
+	U_32 segmentLength = 0;
+	PORT_ACCESS_FROM_JAVAVM(vm);
+	
+	if (NULL == classRep) {
+		throwNewNullPointerException(env, NULL);
+		return;
+	}
+	length = (*env)->GetArrayLength(env, classRep);
+	if (0 == length) {
+		return;
+	}
+
+	segmentLength = ESTIMATE_SIZE(length);
+	segment = (U_8*)j9mem_allocate_memory(segmentLength, J9MEM_CATEGORY_VM_JCL);
+	if (NULL == segment) {
+		return;
```

Comment:
Should throw OOM here.

Reply:
Also, please try and have only one return per function - goto is fine.

Reply:
Added OOM error and fixed the multiple return issue.